lib/ohai/plugins/freebsd/cpu.rb

```diff
@@ -27,21 +27,25 @@
     cpuinfo["flags"] = []
 
     # /var/run/dmesg.boot
-    # CPU: Intel(R) Core(TM) i7-3615QM CPU @ 2.30GHz (3516.61-MHz K8-class CPU)
-    # Origin = "GenuineIntel"  Id = 0x306a9  Family = 6  Model = 3a  Stepping = 9
-    # Features=0x783fbff<FPU,VME,DE,PSE,TSC,MSR,PAE,MCE,CX8,APIC,SEP,MTRR,PGE,MCA,CMOV,PAT,PSE36,MMX,FXSR,SSE,SSE2>
-    # Features2=0x209<SSE3,MON,SSSE3>
-    # AMD Features=0x28100800<SYSCALL,NX,RDTSCP,LM>
-    # AMD Features2=0x1<LAHF>
+    # CPU: Intel(R) Core(TM) i7-4980HQ CPU @ 2.80GHz (2793.59-MHz K8-class CPU)
+    #   Origin="GenuineIntel"  Id=0x40661  Family=0x6  Model=0x46  Stepping=1
+    #   Features=0x783fbff<FPU,VME,DE,PSE,TSC,MSR,PAE,MCE,CX8,APIC,SEP,MTRR,PGE,MCA,CMOV,PAT,PSE36,MMX,FXSR,SSE,SSE2>
+    #   Features2=0x5ed8220b<SSE3,PCLMULQDQ,MON,SSSE3,CX16,SSE4.1,SSE4.2,MOVBE,POPCNT,AESNI,XSAVE,OSXSAVE,AVX,RDRAND>
+    #   AMD Features=0x28100800<SYSCALL,NX,RDTSCP,LM>
+    #   AMD Features2=0x21<LAHF,ABM>
+    #   Structured Extended Features=0x2000<NFPUSG>
+    #   TSC: P-state invariant
 
     File.open("/var/run/dmesg.boot").each do |line|
       case line
       when /CPU:\s+(.+) \(([\d.]+).+\)/
         cpuinfo["model_name"] = $1
         cpuinfo["mhz"] = $2
-      when /Origin = "(.+)"\s+Id = (.+)\s+Stepping = (.+)/
+      when /\s+Origin.*"(.*)".*Family.*0x(\S+).*Model.*0x(\S+).*Stepping.*(\S+)/
```

Comment:
I don't believe checking for the leading whitespace is necessary since we aren't matching from the beginning of the line.